### PR TITLE
[improve][admin] PIP-369 Change default value of `unload-scope` in `ns-isolation-policy set`

### PR DESCRIPTION
**Pull request by grssam**
This PR is a followup to #23120 to make the backward incompatible change to:
* Make the default value of `unload-scope` to `changed`
* Change behavior of all_matching to cover both removed and added regexes

PIP: #23116 

This change is only meant for pulsar 4.x release

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change added tests and can be verified as follows:

  - Added test case to validate behavior of `all_matching` flag
  - Added test case to verify default changed to `changed`

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [x] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [x] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [x] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/grssam/pulsar/pull/3

### CODE REVIEW

**Review by lhotari — APPROVED** [review]
LGTM

**Comment by grssam** [comment]
I just noticed. Why is the codecov diff comparing against a 6 months old commit rather than master's head?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java [hunk 1]
```diff
@@ -802,46 +803,44 @@ private CompletableFuture<Void> filterAndUnloadMatchedNamespaceAsync(String clus
...
+            // list is same, so we continue finding the changed namespaces.
 
-                    log.debug("combined regexes: {}; common regexes:{}", combinedNamespaces, combinedNamespaces);
+            // We create a intersection of the old and new regexes. These won't need to be unloaded.
+            Set<String> commonNamespaces = new HashSet<>(policyData.getNamespaces());
+            commonNamespaces.retainAll(oldNamespaces);
 
-                    // Find the changed regexes (new - new ∩ old). TODO for 4.x, make this (new U old - new ∩ old)
-                    combinedNamespaces.removeAll(commonNamespaces);
+            log.debug("combined regexes: {}; common regexes:{}", combinedNamespaces, combinedNamespaces);
```
Comment (nodece) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java:829]:
```suggestion
            log.debug("combined regexes: {}; common regexes:{}", combinedNamespaces, commonNamespaces);
```

**Review by nodece — APPROVED** [review]

**Pull request merged**